plugin.rb

```diff
@@ -105,27 +105,27 @@ class Engine < ::Rails::Engine
     reaction_users_counting_as_like.flatten!
 
     likes =
-      object
-        .post_actions
-        .select do |post_action|
-          post_action.post_action_type_id == PostActionType.types[:like] && !post_action.trashed?
-        end
-        .reject do |post_action|
-          # Get rid of any PostAction records that match up to a ReactionUser
-          # that is NOT main_reaction_id and is NOT excluded, otherwise we double
-          # up on the count/reaction shown in the UI.
-          is_reaction_like_duplicate =
-            reaction_users_counting_as_like.any? { |ru| ru.user_id == post_action.user_id }
-
-          # Also get rid of any PostAction records that match up to a ReactionUser
-          # that is now the main_reaction_id and has historical data.
-          is_previously_enabled_reaction =
-            post_action.reaction_user.present? &&
-              post_action.reaction_user.reaction.reaction_value ==
-                DiscourseReactions::Reaction.main_reaction_id
-
-          is_reaction_like_duplicate || is_previously_enabled_reaction
-        end
+      object.post_actions.reject do |post_action|
+        # Get rid of any PostAction records that match up to a ReactionUser
+        # that is NOT main_reaction_id and is NOT excluded, otherwise we double
+        # up on the count/reaction shown in the UI.
+        is_reaction_like_duplicate =
+          reaction_users_counting_as_like.any? { |ru| ru.user_id == post_action.user_id }
+
+        # Also get rid of any PostAction records that match up to a ReactionUser
+        # that is now the main_reaction_id and has historical data.
+        is_previously_enabled_reaction =
+          object
+            .post_actions_with_reaction_users
+            &.find do |pa|
+              pa.id == post_action.id &&
+                pa.reaction_user&.reaction&.reaction_value ==
+                  DiscourseReactions::Reaction.main_reaction_id
+            end
+            .present?
+
+        is_reaction_like_duplicate || is_previously_enabled_reaction
+      end
```

Comment:
```suggestion
      object.post_actions.reject do |post_action|
        # Get rid of any PostAction records that match up to a ReactionUser
        # that is NOT main_reaction_id and is NOT excluded, otherwise we double
        # up on the count/reaction shown in the UI.
        is_reaction_like_duplicate =
          reaction_users_counting_as_like.any? { |ru| ru.user_id == post_action.user_id }

        next true if is_reaction_like_duplicate

        # Also get rid of any PostAction records that match up to a ReactionUser
        # that is now the main_reaction_id and has historical data.
        is_previously_enabled_reaction =
          object
            .post_actions_with_reaction_users
            &.find do |pa|
              pa.id == post_action.id &&
                pa.reaction_user&.reaction&.reaction_value ==
                  DiscourseReactions::Reaction.main_reaction_id
            end
            .present?

        is_previously_enabled_reaction
      end
```

A bit of a micro-optimisation but you can avoid work if `is_reaction_like_duplicate` is already true.

Reply:
Ah yeah good point, will change :+1: